ruff_lsp/server.py

```diff
@@ -416,9 +431,50 @@ def code_action(params: CodeActionParams) -> list[CodeAction] | None:
                             title=title,
                             kind=CodeActionKind.QuickFix,
                             data=params.text_document.uri,
-                            edit=_create_workspace_edit(
-                                document, cast(Fix, diagnostic.data)
-                            ),
+                            edit=_create_workspace_edit(document, fix),
+                            diagnostics=[diagnostic],
+                        ),
+                    )
+
+    # Add "Disable for this line" for every diagnostic.
+    if not params.context.only or CodeActionKind.QuickFix in params.context.only:
+        for diagnostic in params.context.diagnostics:
+            if diagnostic.source == "Ruff":
+                noqa_row = cast(DiagnosticData, diagnostic.data)["noqa_row"]
+                if noqa_row is not None:
+                    line = document.lines[noqa_row - 1].rstrip("\r\n")
+                    match = NOQA_REGEX.search(line)
+                    # `foo  # noqa: OLD` -> `foo  # noqa: OLD,NEW`
+                    if match and match.group("codes") is not None:
+                        codes = match.group("codes") + f",{diagnostic.code}"
+                        start, end = match.start("codes"), match.end("codes")
+                        new_line = line[:start] + codes + line[end:]
+                    # `foo  # noqa` -> `foo  # noqa: NEW`
```

Comment:
I think this case might be impossible... since if a line has `foo  # noqa`, the error will by definition be suppressed, right? And so this action wouldn't appear?

Reply:
I managed it using the "useless noqa" lint :)